integrations/terraform/testlib/provision_token_test.go

```diff
@@ -313,3 +313,40 @@ func (s *TerraformSuiteOSS) TestProvisionTokenIAMToken() {
 		},
 	})
 }
+
+func (s *TerraformSuiteOSS) TestProvisionTokenV2Gitlab() {
+	ctx, cancel := context.WithCancel(context.Background())
+	s.T().Cleanup(cancel)
+
+	checkRoleDestroyed := func(state *terraform.State) error {
+		_, err := s.client.GetToken(ctx, "test")
+		if trace.IsNotFound(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	name := "teleport_provision_token.token"
+
+	resource.Test(s.T(), resource.TestCase{
+		ProtoV6ProviderFactories: s.terraformProviders,
+		CheckDestroy:             checkRoleDestroyed,
+		Steps: []resource.TestStep{
+			{
+				Config: s.getFixture("provision_token_v2_gitlab_create.tf"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "kind", "token"),
+					resource.TestCheckResourceAttr(name, "metadata.name", "gitlab-test-terraform"),
+					resource.TestCheckResourceAttr(name, "spec.roles.0", "Bot"),
+					resource.TestCheckResourceAttr(name, "spec.join_method", "gitlab"),
+					resource.TestCheckNoResourceAttr(name, "spec.gitlab.allow.0.environment_protected"),
```

Comment:
Just to make sure that this is working properly for when it is set, could we add:

```
ref_protected = true
```

to the fixture, and 

```go
resource.TestCheckResourceAttr(name, "spec.gitlab.allow.0.ref_protected", "true")
```

to the test checks.